crates/node-core/src/args/database.rs

```diff
@@ -26,6 +30,42 @@ impl DatabaseArgs {
     }
 }
 
+/// clap value parser for [`LogLevel`].
+#[derive(Clone, Debug, Default)]
+struct LogLevelValueParser;
+
+impl TypedValueParser for LogLevelValueParser {
+    type Value = LogLevel;
+
+    fn parse_ref(
+        &self,
+        _cmd: &Command,
+        arg: Option<&Arg>,
+        value: &std::ffi::OsStr,
+    ) -> Result<Self::Value, Error> {
+        let val =
+            value.to_str().ok_or_else(|| Error::raw(ErrorKind::InvalidUtf8, "Invalid UTF-8"))?;
+
+        val.parse::<LogLevel>().map_err(|err| {
+            let arg = arg.map(|a| a.to_string()).unwrap_or_else(|| "...".to_owned());
+            let possible_values = LogLevel::value_variants()
+                .iter()
+                .map(|v| format!("- {:?}: {}", v, v.help_message()))
+                .collect::<Vec<_>>()
+                .join("\n");
+            let msg = format!(
+                "Invalid value '{val}' for {arg}: {err}.\n    Possible values:\n{possible_values}"
+            );
+            clap::Error::raw(clap::error::ErrorKind::InvalidValue, msg)
+        })
+    }
+
+    fn possible_values(&self) -> Option<Box<dyn Iterator<Item = PossibleValue> + '_>> {
+        let values =
+            LogLevel::value_variants().iter().map(|v| PossibleValue::new(v.help_message()));
+        Some(Box::new(values))
```

Comment:
this needs the same code as in the issue ` fn to_possible_value` with possiblevalue name + help